src/main/scala/is/hail/variant/VariantDataset.scala

```diff
@@ -333,11 +333,17 @@ class VariantDatasetFunctions(private val vds: VariantSampleMatrix[Genotype]) ex
     val (paths, types, f) = Parser.parseAnnotationExprs(expr, ec, Some(Annotation.VARIANT_HEAD))
 
     val inserterBuilder = mutable.ArrayBuilder.make[Inserter]
-    val finalType = (paths, types).zipped.foldLeft(vds.vaSignature) { case (vas, (ids, signature)) =>
+    val newType = (paths, types).zipped.foldLeft(vds.vaSignature) { case (vas, (ids, signature)) =>
       val (s, i) = vas.insert(TArray(signature), ids)
       inserterBuilder += i
       s
     }
+    val finalType = if (newType.isInstanceOf[TStruct])
+      paths.foldLeft(newType.asInstanceOf[TStruct]) {
+        case (res, path) => res.setFieldAttributes(path, Map("Number" -> "A"))
```

Comment:
So we actually use this also for VDSs. Because there is no concept of A-arrays, we set this attribute and then rely on it for automatically splitting the array when calling split-multi. Ideally I'd really want an AArray type.